socless/events.py

```diff
@@ -94,215 +142,211 @@ def dedup_hash(self):
-                name=execution_id,
+            _ = stepfunctions_client.start_execution(
+                name=self.metadata.execution_id,
                 stateMachineArn=playbook_arn,
-                input=json.dumps(
-                    {
-                        "execution_id": execution_id,
-                        "artifacts": playbook_input["artifacts"],
-                    }
-                ),
-            )
-            socless_log.info(
-                "Playbook execution started",
-                dict(
-                    meta,
-                    **{"statemachinearn": playbook_arn, "execution_id": execution_id},
-                ),
+                input=json.dumps(playbook_input_as_dict),
             )
+            socless_log.info("Playbook execution started", report.__dict__)
         except Exception as e:
+            report.error = str(e)
             socless_log.error(
                 "Failed to start statemachine execution",
-                dict(
-                    meta,
-                    **{
-                        "statemachinearn": playbook_arn,
-                        "execution_id": execution_id,
-                        "error": f"{e}",
-                    },
-                ),
+                report.__dict__,
             )
-            return {"status": False, "message": f"Error: {e}"}
-        return {
-            "status": True,
-            "message": {
-                "execution_id": execution_id,
-                "investigation_id": investigation_id,
-            },
-        }
+        return report
+
+
+def create_events(event_details: dict, context):
+    """Deduplicate and start playbooks from an intial event or list of event details."""
+    event_details.setdefault("created_at", gen_datetimenow())
+
+    # format events from list or single details dict
+    events_list: List[InitialEvent] = []
+    if isinstance(event_details["details"], list):
+        for details_dict in event_details["details"]:
+            events_list.append(
+                InitialEvent(
+                    details=details_dict,
+                    created_at=event_details["created_at"],
+                    event_type=event_details["event_type"],
+                    playbook=event_details["playbook"],
+                    data_types=event_details["data_types"],
+                    event_meta=event_details["event_meta"],
+                    dedup_keys=event_details["dedup_keys"],
+                )
+            )
+    else:
+        events_list.append(
+            InitialEvent(
+                details=event_details["details"],
+                created_at=event_details["created_at"],
+                event_type=event_details["event_type"],
+                playbook=event_details["playbook"],
+                data_types=event_details["data_types"],
+                event_meta=event_details["event_meta"],
+                dedup_keys=event_details["dedup_keys"],
+            )
+        )
```

Comment:
See the other comment tagged #Ref-InitialEvent.
**tldr** Some of these values should be set to defaults reasonable defaults, as not all use-cases that call create_events pass all of them currently.